plugins/local.js

```diff
@@ -54,6 +54,7 @@ module.exports = {
       var data = buf.toString()
       var peer = ref.parseAddress(data)
       if (peer && peer.key !== sbot.id) {
+        if (config.family === 'IPv6') peer.host = buf.address
```

Comment:
This is the crux of things - if a node is saying 'I am a node' it feels weird to me to use the contents of their message as the IP - why not use the address the packet came from? (From what I understand, its so my node can forward publicly routeable advertisements, which makes sense). With ipv6 link local, addresses are unique to the interface, so you cannot share another nodes address and it be able to connect.

Reply:
I also don’t get it..

https://github.com/ssbc/ssb-server/issues/640#issuecomment-466277322